text/0517-io-os-reform.md

```diff
@@ -1380,9 +1465,9 @@ following interface:
 // TcpStream, which contains both a reader and a writer
 
 impl TcpStream {
-    fn connect<A: ToSocketAddrs>(addr: &A) -> io::Result<TcpStream>;
-    fn peer_addr(&self) -> io::Result<SocketAddr>;
-    fn socket_addr(&self) -> io::Result<SocketAddr>;
+    fn connect<A: ToInetAddrs>(addr: &A) -> io::Result<TcpStream>;
+    fn peer_addr(&self) -> io::Result<InetAddr>;
+    fn inet_addr(&self) -> io::Result<InetAddr>;
```

Comment:
Perhaps we should take this opportunity to rename `inet_addr` to something that is more clearly the other side of `peer_addr`, maybe `local_addr` or `our_addr`?